Change many-to-many labels to belongs-to one list

rathrio opened the pull request:
Addresses #47.

rathrio committed:
Change many-to-many labels to belongs-to one list

rathrio commented on app/scrapers/recipe_scraper.rb:
```diff
@@ -18,7 +18,7 @@ def scrape_image(result)
+    result['recipe']['healthLists']
   end
 
   def scrape_links(result)
```
crap 😸

rathrio committed:
Fix wrong search and replace

rathrio commented:
@simplay please review

simplay commented on app/controllers/lists_controller.rb:
```diff
@@ -0,0 +1,47 @@
+  end
+
+  def destroy
+    @list = current_user.lists.find(params[:id])
```
suggestion: would it make sense to define a `find_list` method in the user model?

simplay commented on app/models/item.rb:
```diff
@@ -1,26 +1,31 @@
 
-  def self.with_labels(label_ids)
-    joins(:items_labels).where('items_labels.label_id IN (?)', label_ids).distinct
+  def self.in_lists(list_ids)
```
i think it would help to add a comment what type of input this method expects: 

```ruby
# @param list_ids [Array<Integer>]
def self.in_lists(list_ids)
...
```

simplay commented on app/models/item.rb:
```diff
@@ -31,8 +36,8 @@ def self.unscraped
 
-  def self.create_from(scraper_result, user:)
-    create!(scraper_result.merge(scraped: true, user: user))
+  def list_name
```
yay, the save-navigation operator :smile_cat:  :+1:

simplay commented:
@rathrio : With this change, I get an odd behavior / runtime error.

What I did: bundling and running the migrations

When I go to a certain list (e.g. Movies), type an arbitrary string and then press enter (to add it), then I get the following error message (shown by a flash message): __List must exist and List can't be blank.__

rathrio committed:
Add doc Item.in_lists

rathrio committed:
Fix background scrape on enter

It will now read the correct session value.

Also moves some session helpers to a separate file.
